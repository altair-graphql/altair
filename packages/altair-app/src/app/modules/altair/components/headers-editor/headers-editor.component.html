<div class="headers-editor">
  <div class="headers-editor__list">
    @for (header of headers(); track $index) {
      <div
        class="headers-editor__list-item"
        [ngClass]="{
          'headers-editor__list-item--disabled': !header.enabled,
        }"
      >
        <label
          nz-checkbox
          [ngModel]="header.enabled"
          (ngModelChange)="
            headerEnabledChange.emit({ index: $index, enabled: $event })
          "
        ></label>
        <div class="set-header-input-column">
          <app-x-input
            class="input"
            placeholder="Header key"
            [ngModel]="header.key"
            (ngModelChange)="headerKeyChange.emit({ index: $index, key: $event })"
            (submitChange)="doneChange.emit()"
          />
        </div>
        <div class="set-header-input-column">
          <app-x-input
            class="input"
            placeholder="Header value"
            [ngModel]="header.value"
            (ngModelChange)="
              headerValueChange.emit({ index: $index, value: $event })
            "
            (submitChange)="doneChange.emit()"
          />
        </div>
        <button
          type="button"
          class="header-input-remove-button"
          (click)="removeHeaderChange.emit($index)"
        >
          &times;
        </button>
      </div>
    }
    <button
      class="btn btn--full-width"
      type="button"
      (click)="addHeaderChange.emit()"
    >
      {{ 'ADD_HEADER_TEXT' | translate }}
    </button>
  </div>
  <div class="headers-editor__notices">
    @if (!isElectron) {
      <div class="headers__notice">
        {{ 'HEADERS_FORBIDDEN_NOTICE' | translate }}
        <a
          href="https://altairgraphql.dev/docs/learn/web-limitations"
          target="_blank"
          rel="noopener"
          >Learn more</a
        >
      </div>
    }
    <div class="headers__notice">
      {{ 'HEADERS_GLOBAL_HEADERS_NOTICE' | translate }}
      <a
        href="https://altairgraphql.dev/docs/features/environment-variables#special-environment-variables"
        target="_blank"
        rel="noopener"
        >Learn more</a
      >
    </div>
  </div>
</div>
