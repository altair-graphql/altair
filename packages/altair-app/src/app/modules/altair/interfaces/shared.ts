export interface IDictionary<V = any> {
  [key: string]: V;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type TODO = any;

export type UnknownError = TODO;

export interface TrackByIdItem {
  id: string;
}
