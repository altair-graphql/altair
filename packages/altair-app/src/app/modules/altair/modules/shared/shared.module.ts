import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DragDropModule } from '@angular/cdk/drag-drop';
import { TranslateModule } from '@ngx-translate/core';
import { NgPipesModule } from 'ngx-pipes';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';
import { MarkdownModule } from 'ngx-markdown';
import { HotToastModule } from '@ngneat/hot-toast';
import { ResizableModule } from 'angular-resizable-element';

import { NzButtonModule } from 'ng-zorro-antd/button';
import { NzFormModule } from 'ng-zorro-antd/form';
import { NzSelectModule } from 'ng-zorro-antd/select';
import { NzModalModule } from 'ng-zorro-antd/modal';
import { NzDropDownModule } from 'ng-zorro-antd/dropdown';
import { NzSwitchModule } from 'ng-zorro-antd/switch';
import { NzListModule } from 'ng-zorro-antd/list';
import { NzRadioModule } from 'ng-zorro-antd/radio';
import { NzAlertModule } from 'ng-zorro-antd/alert';
import { NzLayoutModule } from 'ng-zorro-antd/layout';
import { NzMenuModule } from 'ng-zorro-antd/menu';
import { NzCollapseModule } from 'ng-zorro-antd/collapse';
import { NzInputModule } from 'ng-zorro-antd/input';
import { NzInputNumberModule } from 'ng-zorro-antd/input-number';
import { NzAutocompleteModule } from 'ng-zorro-antd/auto-complete';
import { NzCheckboxModule } from 'ng-zorro-antd/checkbox';
import { NzToolTipModule } from 'ng-zorro-antd/tooltip';
import { NzTabsModule } from 'ng-zorro-antd/tabs';
import { NzTableModule } from 'ng-zorro-antd/table';
import { NzTreeSelectModule } from 'ng-zorro-antd/tree-select';
import { NzGridModule } from 'ng-zorro-antd/grid';
import { NzPopoverModule } from 'ng-zorro-antd/popover';

import { IconsModule } from '../icons/icons.module';
import { NzAvatarModule } from 'ng-zorro-antd/avatar';

const AntdComponentModules = [
  NzButtonModule,
  NzInputModule,
  NzInputNumberModule,
  NzAutocompleteModule,
  NzCheckboxModule,
  NzFormModule,
  NzGridModule,
  NzTabsModule,
  NzSelectModule,
  NzModalModule,
  NzDropDownModule,
  NzSwitchModule,
  NzListModule,
  NzRadioModule,
  NzAlertModule,
  NzLayoutModule,
  NzMenuModule,
  NzCollapseModule,
  NzToolTipModule,
  NzTableModule,
  NzTreeSelectModule,
  NzPopoverModule,
  NzAvatarModule,
  // NgZorroAntdModule,
];

@NgModule({
  imports: [
    CommonModule,
    DragDropModule,
    TranslateModule.forRoot(),
    NgPipesModule,
    NguiAutoCompleteModule, // use antd
    MarkdownModule.forRoot(),
    ResizableModule,
    IconsModule,
    // ToastrModule.forRoot({
    //   newestOnTop: false,
    //   closeButton: true,
    //   positionClass: 'toast-top-center',
    //   enableHtml: true,
    //   countDuplicates: true,
    //   preventDuplicates: true,
    //   resetTimeoutOnDuplicate: true,
    // }),
    // HotToastModule.forRoot({
    //   position: 'top-center',
    //   reverseOrder: true,
    //   dismissible: true,
    //   closeStyle: {
    //     backgroundImage: `none`,
    //   },
    // }),
    ...AntdComponentModules,
  ],

  // DON'T FORGET TO EXPORT MODULE
  exports: [
    DragDropModule,
    TranslateModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    MarkdownModule,
    ResizableModule,
    IconsModule,
    HotToastModule,
    ...AntdComponentModules,
  ],
})
export class SharedModule {
  static forRoot(): ModuleWithProviders<SharedModule> {
    return {
      ngModule: SharedModule,
      providers: [],
    };
  }
}
