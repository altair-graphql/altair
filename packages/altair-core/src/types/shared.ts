export type IDictionary<V = any> = Record<string, V>;

export type TODO = any;

// export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export interface BaseDocument {
  id: string;
  created_at?: number;
  updated_at?: number;
}

export interface BaseOwnableDocument extends BaseDocument {
  ownerUid: string;
}

type TypedOmit<T, K extends keyof T> = Omit<T, K>;
export type CreateDTO<
  T extends BaseDocument | BaseOwnableDocument
> = T extends BaseOwnableDocument
  ? TypedOmit<T, keyof BaseOwnableDocument>
  : TypedOmit<T, keyof BaseDocument>;
export type UpdateDTO<T extends BaseDocument | BaseOwnableDocument> = CreateDTO<
  T
> &
  Pick<T, 'id'>;
