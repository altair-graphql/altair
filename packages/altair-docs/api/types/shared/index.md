# types/shared

## Index

| Member | Description |
| :------ | :------ |
| [BaseDocument](interfaces/BaseDocument.md) | - |
| [BaseOwnableDocument](interfaces/BaseOwnableDocument.md) | - |
| [CreateDTO](type-aliases/CreateDTO.md) | - |
| [IDictionary](type-aliases/IDictionary.md) | - |
| [NoExtraProperties](type-aliases/NoExtraProperties.md) | - |
| [TODO](type-aliases/TODO.md) | - |
| [UpdateDTO](type-aliases/UpdateDTO.md) | - |
