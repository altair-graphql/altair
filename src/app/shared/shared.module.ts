import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { TranslateModule } from '@ngx-translate/core';
import { NgxElectronModule } from 'ngx-electron';
import { SortablejsModule } from 'angular-sortablejs';
import { NgPipesModule } from 'ngx-pipes';
import { ContextMenuModule } from 'ngx-contextmenu';
import { DndModule } from '@beyerleinf/ngx-dnd';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';


@NgModule({
  imports: [
    CommonModule,
    TranslateModule.forRoot(),
    NgxElectronModule,
    SortablejsModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    ContextMenuModule.forRoot(),
    DndModule.forRoot(),
  ],
  exports: [
    TranslateModule,
    NgxElectronModule,
    SortablejsModule,
    NgPipesModule,
    ContextMenuModule,
    DndModule,
    NguiAutoCompleteModule,
  ]
})
export class SharedModule { }
