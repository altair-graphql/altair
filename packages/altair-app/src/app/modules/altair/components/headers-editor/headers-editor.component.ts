import {
  ChangeDetectionStrategy,
  Component,
  input,
  output
} from '@angular/core';
import { HeaderState } from 'altair-graphql-core/build/types/state/header.interfaces';
import { isElectronApp } from '../../utils';

@Component({
  selector: 'app-headers-editor',
  templateUrl: './headers-editor.component.html',
  styles: ``,
  changeDetection: ChangeDetectionStrategy.OnPush,
  standalone: false,
})
export class HeadersEditorComponent {
  readonly headers = input<HeaderState>([]);

  readonly headerEnabledChange = output<{
    index: number;
    enabled: boolean;
}>();
  readonly headerKeyChange = output<{
    index: number;
    key: string;
}>();
  readonly headerValueChange = output<{
    index: number;
    value: string;
}>();
  readonly addHeaderChange = output();
  readonly removeHeaderChange = output<number>();
  readonly doneChange = output();

  isElectron = isElectronApp();
}
