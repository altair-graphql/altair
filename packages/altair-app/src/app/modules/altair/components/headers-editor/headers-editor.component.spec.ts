import { HeadersEditorComponent } from './headers-editor.component';
import { mount, NgxTestWrapper } from '../../../../../testing';
import { MockModule } from 'ng-mocks';
import { SharedModule } from '../../modules/shared/shared.module';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('HeadersEditorComponent', () => {
  let wrapper: NgxTestWrapper<HeadersEditorComponent>;

  beforeEach(async () => {
    wrapper = await mount({
      component: HeadersEditorComponent,
      imports: [MockModule(SharedModule)],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  it('should create', () => {
    expect(wrapper.componentInstance).toBeTruthy();
  });

  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
