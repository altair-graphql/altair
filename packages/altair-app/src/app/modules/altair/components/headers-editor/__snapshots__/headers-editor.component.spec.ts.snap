// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadersEditorComponent should render correctly 1`] = `
<app-headers-editor>
  <div
    class="headers-editor"
  >
    <div
      class="headers-editor__list"
    >
      
      <button
        class="btn btn--full-width"
        type="button"
      >
          
      </button>
    </div>
    <div
      class="headers-editor__notices"
    >
      <div
        class="headers__notice"
      >
          
        <a
          href="https://altairgraphql.dev/docs/learn/web-limitations"
          rel="noopener"
          target="_blank"
        >
          Learn more
        </a>
      </div>
      
      <div
        class="headers__notice"
      >
          
        <a
          href="https://altairgraphql.dev/docs/features/environment-variables#special-environment-variables"
          rel="noopener"
          target="_blank"
        >
          Learn more
        </a>
      </div>
    </div>
  </div>
</app-headers-editor>
`;
